lib/plugin-compat.php

```diff
@@ -25,10 +25,12 @@
  * @param  array $post      Post object which contains content to check for block.
  * @return array $post      Post object.
  */
-function gutenberg_remove_wpcom_markdown_support( $post ) {
-	if ( gutenberg_content_has_blocks( $post->post_content ) ) {
-		remove_post_type_support( 'post', 'wpcom-markdown' );
+function gutenberg_remove_wpcom_markdown_support( $post_data ) {
+	if ( gutenberg_content_has_blocks( implode( "\n", $post_data ) ) ) {
+        if ( class_exists ( 'WPCom_Markdown' ) ) {
+            WPCom_Markdown::get_instance()->unload_markdown_for_posts();
+        }
 	}
-	return $post;
+	return $post_data;
 }
-add_filter( 'rest_pre_insert_post', 'gutenberg_remove_wpcom_markdown_support' );
+add_filter( 'wp_insert_post_data', 'gutenberg_remove_wpcom_markdown_support', 1, 1 );
```

Comment:
Needs to be so early in priority? Can't it just be one step before where Jetpack it filtering it at?

Reply:
The fourth argument is `1` by default and can be omitted.

Reply:
Updated.